### ISSUE TITLE
UseEffect() rendering the routes - Twice

### ISSUE THREAD

**Issue by ChandanKhamitkar** [Chandan-6/Nova-Bills#1]
Fix Bug **GH-6a12134**

Here, in the route, an array list of data is rendered using GET-request. But in the **useEffect()** calling the route **twice**, which causing the below error.
```javascript
  const [invoiceData, setInvoiceData] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const token = Cookies.get("nb_token");
        const response = await axios.get(
          "http://localhost:8000/api/user/getInvoice",
          {
            headers: {
              "content-type": "application/json",
              Authorization: `Bearer ${token}`,
            },
          }
        );

        if (response.data.success) {
          setInvoiceData(response.data.invoiceData);
        } else {
          toast.error("Error in retriving data!");
        }
      } catch (error) {
        console.log(error);
        toast.error("Internal server error!");
      }
    };

    fetchData();
  }, []);
``` 
Due to which causing **Internal Server Error**

![Screenshot 2024-04-19 190539](https://github.com/Chandan-6/Nova-Bills/assets/141998868/a4646307-2ba1-4160-b24c-5c620660c940)

**Comment by Chandan-6** [comment]
Fixed by #2 

**Fix 1** - In the above code, the useEffect() dependencies are empty, so it will render for every change in the page.
**Fix 2** - Altered to only one response in route rather than 2 responses, which causing **can't set headers**

_Fix 1 - ( useEffect rendering twice issue )_
_Fix 2 - ( can't set headers issue)_

**Comment by ChandanKhamitkar** [comment]
Fixed by #2 

**Fix 1** - In the above code, the useEffect() dependencies are empty, so it will render for every change in the page.
**Fix 2** - Altered to only one response in route rather than 2 responses, which causing **can't set headers**

_Fix 1 - ( useEffect rendering twice issue )_
_Fix 2 - ( can't set headers issue)_
